src/util/psbt/mod.rs

```diff
@@ -208,6 +217,471 @@ impl PartiallySignedTransaction {
 
         Ok(())
     }
+
+    /// Attempts to create _all_ the required signatures for this PSBT using `k`.
+    ///
+    /// If you just want to sign an input with one specific key consider using `sighash_ecdsa`. This
+    /// function does not support scripts that contain `OP_CODESEPARATOR`.
+    ///
+    /// # Panics
+    /// If an attempt is made to sign a taproot input.
+    pub fn sign<C: Signing>(&mut self, k: impl GetKey + Copy, secp: &Secp256k1<C>) -> Result<(), SignError> {
+        let tx = self.unsigned_tx.clone();
```

Comment:
Just making sure that this clone is necessary. It could be because mutable borrow is required in the cache. If it is difficult to remove this, we should add a comment explaining why.

Reply:
Signing the PSBT (`self`) needs a mutable borrow below but passes in the cache, so the cache can't have an immutable borrow at the same time. I added a comment.